src/lib/transition_frontier_controller/transition_frontier_controller.ml

```diff
@@ -6,17 +6,18 @@ open O1trace
 let run ~logger ~trust_system ~verifier ~network ~time_controller
     ~collected_transitions ~frontier ~network_transition_reader
     ~producer_transition_reader ~clear_reader ~precomputed_values =
-  let valid_transition_pipe_capacity = 30 in
+  let valid_transition_pipe_capacity = 50 in
   let valid_transition_reader, valid_transition_writer =
     Strict_pipe.create ~name:"valid transitions"
-      (Buffered (`Capacity valid_transition_pipe_capacity, `Overflow Crash))
+      (Buffered (`Capacity valid_transition_pipe_capacity, `Overflow Drop_head))
   in
   let primary_transition_pipe_capacity =
     valid_transition_pipe_capacity + List.length collected_transitions
   in
+  (* Ok to drop on overflow- catchup will be triggered if required*)
   let primary_transition_reader, primary_transition_writer =
     Strict_pipe.create ~name:"primary transitions"
-      (Buffered (`Capacity primary_transition_pipe_capacity, `Overflow Crash))
+      (Buffered (`Capacity primary_transition_pipe_capacity, `Overflow Drop_head))
```

Comment:
When these transitions are dropped from the pipes, we should probably fill the validation callback for libp2p's sake and tell it that to ignore that message. This will make the logs more useful since we will filter out any validation callback timeouts caused by dropping these transitions. You can provide a function to call when messages are dropped from the pipe by using `Overflow (Call f)`.

Reply:
There are warning logs with the `Drop_head option.  `Call f` doesn't drop but just executes f on the data it is trying to write. Are you suggesting we drop the latest message instead of oldest message that gets removed with the `Drop_head` variant?